### PR TITLE
[Yaml] fix PHP const mapping keys using the inline notation

### LINKED ISSUE #35179: Using a PHP class constant for a key in a YAML hash as a method argument produces an ErrorException
**Symfony version(s) affected**: 4.4.2

**Description**  
When trying to use a PHP class constant as a YAML key for a hash (as a method argument), I receive an `ErrorException` with the message `Uninitialized string offset: 0` and code `8`.

**How to reproduce**
Using the following class and Symfony YAML and DI I receive an `ErrorException` with the message `Uninitialized string offset: 0` and code `8`.
```php
<?php
declare(strict_types=1);

namespace VENDOR\PRODUCT;

class Example
{
    public const CONSTANT_1 = 'constant_1';
    public const CONSTANT_2 = 'constant_2';

    private $Options;

    public function addOptions(array $options): Example
    {
        $this->Options[] = $options;

        return $this;
    }
}
```
```yml
services:
  VENDOR\PRODUCT\Example:
    class: VENDOR\PRODUCT\Example
    public: false # probably unrelated
    shared: true # probably unrelated
    calls:
      - [addOptions, [{
          !php/const 'VENDOR\PRODUCT\Example::CONSTANT_1': !php/const VENDOR\PRODUCT\Example::CONSTANT_2,
          key_1: "value_1",
          key_2: "value_2"
        }]]
```

**Possible Solution**  

**Additional context** 
I am more than willing to bet that I am formatting the YAML incorrectly to accomplish this, but I have tried many variations of indentations and YAML hash expressions. Constants as YAML hash keys *do* work if I use them like the following. So my assumption of the behavior is that they should be able to be used for any valid YAML hash expression (like my example above).

```yml
    arguments:
      - !php/const '\VENDOR\PRODUCT\Example::CONSTANT_1': !php/const \VENDOR\PRODUCT\Example::CONSTANT_2
```

I have quoted the constant for the key in my example above due to [this](https://github.com/symfony/symfony/pull/22913#issuecomment-303979423) comment. However, I have tried it without quoting and I receive an error about the colons (which I expect due to YAML processing).

Thank you =)

***Update***
I should probably also mention that the control case works as expected, i.e.

```yml
services:
  VENDOR\PRODUCT\Example:
    class: VENDOR\PRODUCT\Example
    public: false # probably unrelated
    shared: true # probably unrelated
    calls:
      - [addOptions, [{
          key_0: "value_0",
          key_1: "value_1",
          key_2: "value_2"
        }]]
```

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #35179
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Comment by rhift** [comment]
hey @xabbuh will this make its way into the YAML component? Sorry, I'm not familiar with the release process. Thanks for fixing it!

**Comment by xabbuh** [comment]
@rhift Yes, this change will be part of the next patch releases for 3.4, 4.3, 4.4 and 5.0 as well as future minor versions.

File: src/Symfony/Component/Yaml/Inline.php [hunk 1]
```diff
@@ -504,6 +504,11 @@ private static function parseMapping($mapping, $flags, &$i = 0, $references = []
             $isKeyQuoted = \in_array($mapping[$i], ['"', "'"], true);
             $key = self::parseScalar($mapping, $flags, [':', ' '], $i, false, [], true);
 
+            if ('!php/const' === $key) {
```
Comment (fabpot) [inline src/Symfony/Component/Yaml/Inline.php]: Shouldn't we check for `self::$constantSupport` first? /cc @xabbuh

Comment (xabbuh) [reply src/Symfony/Component/Yaml/Inline.php]: I think this is fine because `evaluateScalar()` (which is called now) does the check.

**Comment by rhift** [comment]
Hi there, thanks for this!

I can get globally defined constants to work great, but I can't find a way to get class constants to work. Is this intended behavior?

I've tried the following.
```yml
!php/const \PATH\TO\Class::CONSTANT: "value"
!php/const "\PATH\TO\Class::CONSTANT": "value"
```

The scope resolution operator (`::`) is trimmed and the constant cannot be found.
 
Being able to use class constants as keys was what I was hoping for when submitting https://github.com/symfony/symfony/issues/35179.

Any help would be appreciated. Thank you. =)

**Comment by xabbuh** [comment]
@rhift If I don't miss anything, we have these cases covered with tests. Could you open a new issue please with some code that allows to reproduce so that we can see whether there is another bug or if there is something else going wrong?

**Comment by rhift** [comment]
@xabbuh ok, thanks! I opened https://github.com/symfony/symfony/issues/36624.

**Pull request merged**